File: app/src/main/java/com/shenhua/zhidaodaily/utils/JSInterface.java
package com.shenhua.zhidaodaily.utils;

import android.content.Context;
import android.content.Intent;
import android.text.TextUtils;
import android.webkit.JavascriptInterface;
import android.widget.Toast;

import com.shenhua.zhidaodaily.ui.PhotoActivity;

/**
 * 通用WebView JS interface
 * Created by Shenhua on 12/10/2016.
 * e-mail shenhuanet@126.com
 */
public class JSInterface {
    private Context context;

    public JSInterface(Context context) {
        this.context = context;
    }

    @JavascriptInterface
    public void openImage(String url) {
        if (url != null && !TextUtils.isEmpty(url)) {
            if (url.endsWith("gif")) {
                Toast.makeText(context, "暂不能打开gif", Toast.LENGTH_SHORT).show();
                return;
            }
            Intent intent = new Intent(context, PhotoActivity.class);
            intent.putExtra("imgUrl", url);
            context.startActivity(intent);
        }
    }
}

File: app/src/main/java/com/shenhua/zhidaodaily/utils/JsInterface.kt
package com.shenhua.zhidaodaily.utils

import android.os.Bundle
import android.text.TextUtils
import android.view.View
import android.webkit.JavascriptInterface
import android.widget.Toast
import androidx.navigation.Navigation
import com.shenhua.zhidaodaily.R

/**
 * 通用WebView JS interface
 * Created by Shenhua on 12/10/2016.
 * e-mail shenhuanet@126.com
 *
 * @author shenhua
 */
class JsInterface(private val view: View) {

    /**
     * JavascriptInterface 此注解一定要加上
     *
     * @param url url
     */
    @JavascriptInterface
    fun openImage(url: String?) {
        if (url != null && !TextUtils.isEmpty(url)) {
            val bundle = Bundle()
            bundle.putString("image", url)
            Navigation.findNavController(view).navigate(R.id.action_contentFragment_to_photoFragment, bundle)
        }
    }

    @JavascriptInterface
    fun loadHtmlContent(content: String) {
        if (TextUtils.isEmpty(content)) {
            Toast.makeText(view.context, "暂无可分享内容", Toast.LENGTH_SHORT).show()
            return
        }
        Utils.shareText(view.context, "来自知道日报: \n$content")
    }
}